client/ayon_core/hosts/houdini/plugins/create/create_bgeo.py

```diff
@@ -61,45 +60,56 @@ def create(self, product_name, instance_data, pre_create_data):
 
         instance_node.setParms(parms)
 
+    @staticmethod
+    def update_node_parameters(node, creator_attributes):
+        """update node parameters according to creator attributes.
+
+        Implementation of update_node_parameters.
+
+        Args:
+            node(hou.Node): Houdini node to apply changes to.
+            creator_attributes(dict): Dictionary of creator attributes.
+        """
+
+        file_path, _ = lib.splitext(
+            node.evalParm("sopoutput"),
+            allowed_multidot_extensions=[
+                ".ass.gz", ".bgeo.sc", ".bgeo.gz",
+                ".bgeo.lzma", ".bgeo.bz2"
+            ]
+        )
+
+        output = "{file_path}.{ext}".format(
+            file_path=file_path,
+            ext=creator_attributes["bgeo_type"]
+        )
+
+        node.setParms({"sopoutput": output})
+
     def get_instance_attr_defs(self):
+        bgeo_enum = {
+            "bgeo": "uncompressed bgeo (.bgeo)",
+            "bgeosc": "BLOSC compressed bgeo (.bgeosc)",
+            "bgeo.sc": "BLOSC compressed bgeo (.bgeo.sc)",
+            "bgeo.gz": "GZ compressed bgeo (.bgeo.gz)",
+            "bgeo.lzma": "LZMA compressed bgeo (.bgeo.lzma)",
+            "bgeo.bz2": "BZip2 compressed bgeo (.bgeo.bz2)",
+        }
+
         return [
             BoolDef("farm",
                     label="Submitting to Farm",
-                    default=False)
+                    default=False),
+            EnumDef("bgeo_type",
+                    items=bgeo_enum,
+                    default="bgeo",
+                    label="BGEO Options")
         ]
 
     def get_pre_create_attr_defs(self):
         attrs = super().get_pre_create_attr_defs()
-        bgeo_enum = [
-            {
-                "value": "bgeo",
-                "label": "uncompressed bgeo (.bgeo)"
-            },
-            {
-                "value": "bgeosc",
-                "label": "BLOSC compressed bgeo (.bgeosc)"
-            },
-            {
-                "value": "bgeo.sc",
-                "label": "BLOSC compressed bgeo (.bgeo.sc)"
-            },
-            {
-                "value": "bgeo.gz",
-                "label": "GZ compressed bgeo (.bgeo.gz)"
-            },
-            {
-                "value": "bgeo.lzma",
-                "label": "LZMA compressed bgeo (.bgeo.lzma)"
-            },
-            {
-                "value": "bgeo.bz2",
-                "label": "BZip2 compressed bgeo (.bgeo.bz2)"
-            }
-        ]
 
-        return attrs + [
-            EnumDef("bgeo_type", bgeo_enum, label="BGEO Options"),
-        ] + self.get_instance_attr_defs()
+        return attrs + self.get_instance_attr_defs()
```

Comment:
Should the `farm` toggle now also be exposed on pre create?

Reply:
tbh, I'm not sure where the `farm` value should exist: 
- should it be a checkbox parameter
- or should it be a `creator_attribute`

Reply:
I believe in the near future, farm checkbox will be removed in favor of putting it inside creator attribute.
As, for consistency we may want to add render targets for caches too like https://github.com/ynput/ayon-core/pull/328 
So, let's ignore farm toggle atm.
Feel free to mark this back as unresolved.

Reply:
> or should it be a creator_attribute

I'd expect it's a toggle on instance attribute defs (and potentially _already_ available to preset on the creator attributes) so that at any point in time one can switch between farm submission and local render - similar to how Fusion an Nuke are doing it with "render local" versus "farm" enum options.

Reply:
I think there are two problems: 
1. Which parameter should be the source of truth ?
2. How to update related to parameters to follow the source of truth ?

Let's take redshift creator plugin as an example: 
a. In `pre_create_defs`, we add a `BoolDef("farm",...)` to add a toggle to the creator plugin. 
b. In `process`, we add the value of that def in a farm "toggle"
c. At this point the value of the `BoolDef("farm",...)` is saved to two parameters (a `toggle` parameter and `creator_attribtes` parameter)
d. In publisher, I can switch off `Submitting to Farm` which will affect the `"farm"` key in `creator_attributes`. which basically does nothing, and it doesn't affect the farm `toggle`. 
e. I can only switch off the `farm` publishing by clicking the farm `toggle` manually.

So, all plugins expects the farm toggle to be the source of truth (which automatically collected by imprint) and the same with `splitRender` feature which is based on the parameter `RS_archive_enable`  (which is not exposed in the publisher)
at this point the two issues above are not exposed yet.

But, these issues started to show up in https://github.com/ynput/ayon-core/pull/328 when I started to use `get_instance_attr_defs` which allows me to change `creator_attributes` in publish tab.
And, The `render_target` and render mode were exposed in the publisher. 

So, what parameters should be the source of truth. 
- `creator_attributes` parameter which acts as a variable that holds the last selection
- or the actual Houdini parameters that get updated eventually based on the selection in the publisher. 

Which is as you said earlier 
```
but I'd also then try to "read" the state of that publisher value from that parameter instead of storing
 other 'metadata' on the node that gets out of sync.
```

Then let's think of the second question:  How to update related to parameters to follow the source of truth ?
if the Houdini parameters are the source of truth then we should have some inverted logic to deduce the correct `creator_attributes` so that they show up in the publisher correctly which _I assume_ should work on `refresh` action. 

and when opening the publisher we use _maybe this PR_ to apply the changes in the publisher to the instance node. 

I started to feel that I'm overthinking but tbh I consider this a bad issue.
as it touches many problems in other PRs. 
e.g. this one https://github.com/ynput/ayon-core/pull/328#discussion_r1558137542 
because I don't feel it should exist in a collector or extractor. 
The collector should only collect the data. 
The extractor can do any changes but it should revert the changes
However, we want just to update the instance according to our settings and me as a user expected the changes to be applied once I hit save similar to the changes happen in the creator.

Reply:
maybe this commit adding the missing piece of the puzzle https://github.com/ynput/ayon-core/pull/418/commits/101a86d386c7b3812edcd2477aca1fc3dbe646b0

Reply:
> Should the farm toggle now also be exposed on pre create?

yes, implemented in https://github.com/ynput/ayon-core/pull/418/commits/b865d49f23d44dc21abb8d9202b966955e5813e6